### PR TITLE
[java] Fix processing errors

### PR DESCRIPTION
**Pull request by adangel**
I was trying to reproduce #650 , no luck yet.
However, I've added the check for parsing/processing errors in PMDCoverageTest, and started to fix some. I'm not finished yet, but wanted to share the current status.

Note: These are all bugs found while analyzing PMD's own code with PMD... 

There are currently 41 processing errors left...

### CODE REVIEW

**Comment by jsotuyod** [comment]
@adangel very nice! I was looking into diving into this myself, so let me know if you need help.

For the cases where typeresolution is still not complete, we may want just to return null as type (as we used to), just avoiding the ProcessingError. Let me know if I can help you tackle this.

**Comment by adangel** [comment]
This should be done now. There are no processing errors left and no existing unit test started to fail - so I hope, I didn't make it worse :)

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java [hunk 1]
```diff
@@ -603,6 +624,10 @@ public static boolean isMethodConvertible(JavaTypeDefinition parameter, JavaType
 
 
     public static boolean isSubtypeable(JavaTypeDefinition parameter, ASTExpression argument) {
+        if (argument.getTypeDefinition() == null) {
+            LOG.log(Level.FINE, "No type information for node {0}", argument.toString());
+            return true;
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java]: is true a good default? shouldn't it be false upon unknown types?

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java [hunk 2]
```diff
@@ -564,6 +581,10 @@ public static int getArity(Method method) {
     }
 
     public static boolean isMethodConvertible(JavaTypeDefinition parameter, ASTExpression argument) {
+        if (argument.getTypeDefinition() == null) {
+            LOG.log(Level.FINE, "No type information for node {0}", argument.toString());
+            return true;
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java]: same here, shouldn't we default to false?

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java (hunk 1 above)
Comment (adangel) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java]: I think, I tried it - and got not so good results: In that case, we might not find the correct method call...on the other hand, we might match a wrong method call...

It's also similar to https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java#L619 - where we originally got a NPE.

Comment (jsotuyod) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java]: sounds reasonable, thanks!

**Comment by adangel** [comment]
It very likely, that #534 and #510 are fixed - I came across the same exceptions. For #650 I'm not sure, since we don't have a stacktrace there... (which reminds me, to check, whether this is maven-pmd-plugin problem or pmd problem)

**Pull request merged**